include/cache.php

```diff
@@ -84,9 +84,14 @@ public static function get($key) {
 		$memcache = self::memcache();
 		if (is_object($memcache)) {
 			// We fetch with the hostname as key to avoid problems with other applications
-			$value = $memcache->get(get_app()->get_hostname().":".$key);
-			if (!is_bool($value)) {
-				return unserialize($value);
+			$cached = $memcache->get(get_app()->get_hostname().":".$key);
+			$value = @unserialize($cached);
+
+			// Only return a value if the serialized value is valid.
+			// We also check if the db entry is a serialized
+			// boolean 'false' value (which we want to return).
+			if ($cached === 'b:0;' || $value !== false) {
+				return $value;
```

Comment:
@annando Then in that case I suggest we use `json_decode` for DB cache storing. It is faster than `unserialize`, JSON strings are shorter than serialized strings, and it returns `null` if it couldn't decode the stored value, which would automatically trigger the processing script in the Memcache example I wrote earlier.

Overall changes:
- Remove `serialize`/`unserialize` functions for Memcache
- Replace `serialize`/`unserialize` functions by `json_encode`/`json_decode` respectively for DB cache